extensions/spark/kyuubi-spark-connector-hive/src/test/scala/org/apache/kyuubi/spark/connector/hive/command/CreateNamespaceSuite.scala

```diff
@@ -0,0 +1,139 @@
+    val ns = s"$catalogName.$namespace"
+    withNamespace(ns) {
+      withTempDir { tmpDir =>
+        // The generated temp path is not qualified.
+        val path = tmpDir.getCanonicalPath
+        assert(!path.startsWith("file:/"))
+
+        val e = intercept[IllegalArgumentException] {
+          sql(s"CREATE NAMESPACE $ns LOCATION ''")
+        }
+        assert(e.getMessage.contains("Can not create a Path from an empty string"))
+
+        val uri = new Path(path).toUri
+        sql(s"CREATE NAMESPACE $ns LOCATION '$uri'")
+
+        // Make sure the location is qualified.
+        val expected = makeQualifiedPath(tmpDir.toString)
+        assert("file" === expected.getScheme)
+        assert(new Path(getNamespaceLocation(catalogName, namespaceArray)).toUri === expected)
+      }
+    }
+  }
+
+  test("Namespace already exists") {
+    val ns = s"$catalogName.$namespace"
+    withNamespace(ns) {
+      sql(s"CREATE NAMESPACE $ns")
+
+      val e = intercept[NamespaceAlreadyExistsException] {
+        sql(s"CREATE NAMESPACE $ns")
+      }
+      assert(e.getMessage.contains(s"$notFoundMsgPrefix '$namespace' already exists"))
+
+      // The following will be no-op since the namespace already exists.
+      Try { sql(s"CREATE NAMESPACE IF NOT EXISTS $ns") }.isSuccess
+    }
+  }
+
+  test("CreateNameSpace: reserved properties") {
+    import SupportsNamespaces._
+    val ns = s"$catalogName.$namespace"
+    withSQLConf((SQLConf.LEGACY_PROPERTY_NON_RESERVED.key, "false")) {
+      NAMESPACE_RESERVED_PROPERTIES.filterNot(_ == PROP_COMMENT).foreach { key =>
+        val exception = intercept[ParseException] {
+          sql(s"CREATE NAMESPACE $ns WITH DBPROPERTIES('$key'='dummyVal')")
+        }
+        assert(exception.getMessage.contains(s"$key is a reserved namespace property"))
+      }
+    }
+    withSQLConf((SQLConf.LEGACY_PROPERTY_NON_RESERVED.key, "true")) {
+      NAMESPACE_RESERVED_PROPERTIES.filterNot(_ == PROP_COMMENT).foreach { key =>
+        withNamespace(ns) {
+          sql(s"CREATE NAMESPACE $ns WITH DBPROPERTIES('$key'='foo')")
+          assert(
+            sql(s"DESC NAMESPACE EXTENDED $ns")
+              .toDF("k", "v")
+              .where("k='Properties'")
+              .where("v=''")
+              .count == 1,
+            s"$key is a reserved namespace property and ignored")
+          val meta =
+            getCatalog(catalogName).asNamespaceCatalog.loadNamespaceMetadata(namespaceArray)
+          assert(
+            meta.get(key) == null || !meta.get(key).contains("foo"),
+            "reserved properties should not have side effects")
+        }
+      }
+    }
+  }
+
+  protected def getNamespaceLocation(catalog: String, namespace: Array[String]): String = {
+    val metadata = getCatalog(catalog).asNamespaceCatalog
+      .loadNamespaceMetadata(namespace).asScala
+    metadata(SupportsNamespaces.PROP_LOCATION)
+  }
+}
+
+class CreateNamespaceSuite extends CreateNamespaceSuiteBase {
```

Comment:
Thanks for building this framework, the purpose is to make sure that the V2 implementation should work as same(or have meaningful different behaviors) as the Spark built-in one, could you please test that as well?